packages/svelte/src/internal/client/dom/blocks/await.js

```diff
@@ -71,13 +75,34 @@ export function await_block(anchor, get_input, pending_fn, then_fn, catch_fn) {
 
 		if (is_promise(input)) {
 			const promise = /** @type {Promise<any>} */ (input);
+			var resolved = false;
 
 			if (pending_fn) {
-				if (pending_effect && (pending_effect.f & INERT) === 0) {
-					destroy_effect(pending_effect);
+				var parent_effect = current_effect;
+				var parent_reaction = current_reaction;
+				var show_pending = () => {
+					if (resolved) return;
+					if (pending_effect && (pending_effect.f & INERT) === 0) {
+						destroy_effect(pending_effect);
+					}
+					var previous_effect = current_effect;
+					var previous_reaction = current_reaction;
+					try {
+						set_current_effect(parent_effect);
+						set_current_reaction(parent_reaction);
```

Comment:
why doe we need to set the current effect/reaction? Seems like it wasn't necessary before.